src/deploy/functions/build.ts

```diff
@@ -586,3 +595,32 @@ function discoverTrigger(endpoint: Endpoint, region: string, r: Resolver): backe
   }
   assertExhaustive(endpoint);
 }
+
+function interpolateEventArcChannel(channel: string, projectId: string): string {
```

Comment:
This is largely a copy-paste from the removed function in v1alpha1.ts, with two changes I'd like to draw attention to:
1) the original function accepted defaultRegion as a parameter, which turns out to always be populated by calling `api.functionsDefaultRegion` if you follow the call chain up.
2) the projectId was once provided directly from a variable passed into the discovery process by the runtime delegate. replacing it with `endpoint.project` is safe in this context because `endpoint.project` is always populated from that same variable in v1alpha1.ts.